src/config.rs

```diff
@@ -625,17 +625,37 @@ impl Cfg {
             targets: &[&str],
         ) -> Result<bool> {
             let components_requested = !components.is_empty() || !targets.is_empty();
-
+            // If we're here, the toolchain exists on disk and is a dist toolchain
+            // so we should attempt to load its manifest
+            let manifest = if let Some(manifest) = distributable.get_manifest()? {
+                manifest
+            } else {
+                // If we can't read the manifest we'd best try and install
+                return Ok(true);
```

Comment:
will this remove the corrupt dist?

Reply:
No it'll bubble the situation up a level which will then try to ensure the component set in question and report something along the lines of:

```
error: toolchain 'nightly-2020-11-24-x86_64-unknown-linux-gnu' does not support components
```

I think this is fair enough.

Reply:
Except that that should support components, and the reason it doesn't is that its corrupt, and we know its corrupt because we expected it to support components, and we returned Ok(true) rather than signalling the impossible situation...

I think this is good enough to do a release with; I don't think its good enough to consider done.

Reply:
Actually no, I got that entirely wrong - argh.  Let me faff.

Reply:
I think when I tested this, I must have accidentally reinstalled the test release rather than the version I was editing.  I need to push a change to make that `Ok(false)` and then instead of being crap about it, it'll heal the installation.